projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -67,6 +70,18 @@ def initialize_job(self, context: JobContext) -> None:
             lambda: _connection_by_configuration(self._impala_cfg),
         )
 
+    @hookimpl(tryfirst=True)
+    def vdk_exception(self, exception: Exception):
```

Comment:
Hooking to vdk_exception may not work since classification happens earlier. This way the termination message will not be udpated correctly. 

See how I've done it for trino: https://github.com/vmware/versatile-data-kit/pull/650/files